lib/prepare_release.js

```diff
@@ -533,13 +535,37 @@ export default class ReleasePreparation extends Session {
     const { newVersion } = this;
     const proposalBranch = `v${newVersion}-proposal`;
 
-    await runAsync('git', [
-      'checkout',
-      '-b',
+    try {
+      await forceRunAsync('git', [
+        'checkout',
+        '-b',
+        proposalBranch,
+        base
+      ], { captureStdout: true, captureStderr: true, ignoreFailure: false });
+    } catch (err) {
+      const branchExistsRE = /fatal: a branch named '.*' already exists/i;
+      if (branchExistsRE.test(err.stderr)) {
+        await runAsync('git', [
+          'checkout',
+          proposalBranch
+        ]);
+        await runAsync('git', [
+          'reset',
+          '--hard',
+          base
+        ]);
+        return {
+          proposalBranch,
+          resultMsg: `Updated proposal branch for ${newVersion}`
+        };
+      } else {
+        throw err;
+      }
+    }
```

Comment:
Looking at https://git-scm.com/docs/git-checkout#Documentation/git-checkout.txt-emgitcheckoutem-b-Bltnew-branchgtltstart-pointgt, it looks like we can simplify all this
```suggestion
    await forceRunAsync('git', [
        'checkout',
        '-B',
        proposalBranch,
        base
      ], { ignoreFailure: false });
    }
```

Reply:
Given this part I was not sure that it would work:

> the branch is not reset/created unless "git checkout" is successful (e.g., when the branch is in use in another worktree, not just the current branch stays the same, but the branch is not reset to the start-point, either).

but I just tried replacing with your suggestion here and it works as a charm! so scrap the initial implementation here, I'll go with this 😊